app/controllers/sign_up/email_confirmations_controller.rb

```diff
@@ -18,6 +20,15 @@ def create
 
     private
 
+    def validate_request_id
+      result = RequestIdValidator.new(params[:_request_id]).submit
+      analytics.track_event(Analytics::EMAIL_CONFIRMATION_REQUEST_ID, result.to_h)
+
+      return if result.success?
+      flash[:error] = t('errors.messages.request_id_invalid')
+      redirect_to sign_up_email_resend_url
```

Comment:
How about "Oops, something went wrong. Please try clicking the link in your confirmation email again."